code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -5,13 +5,52 @@
 
 /obj/structure/closet/crate/necropolis
 	name = "necropolis chest"
-	desc = "It's watching you closely."
+	desc = "It's watching you suspiciously."
 	icon_state = "necrocrate"
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	door_anim_time = 0
+	var/need_key = FALSE
+
+/obj/structure/closet/crate/necropolis/update_icon()
+	. = ..()
+	if(need_key == TRUE)
+		add_overlay("necrocrate_lock")
+
+/obj/structure/closet/crate/necropolis/proc/update_desc()
+	desc = "It's watching you suspiciously."
+	if(need_key == TRUE)
+		desc += " It seems like it needs a key to unlock..."
+
+/obj/structure/closet/crate/necropolis/Initialize()
+	. = ..()
+	update_desc()
+
+/obj/structure/closet/crate/necropolis/attack_hand(mob/user)
+	if(need_key == TRUE)
+		to_chat(user, "<span class='warning'>It seems that this chest is locked!</span>")
+		return
+	. = ..()
+
+/obj/structure/closet/crate/necropolis/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(W.type == /obj/item/chest_key)
+		need_key = FALSE
+		update_icon()
+		update_desc()
+		qdel(W)
+		visible_message("The [src] unlocks, destroying the key in the process!")
+
+/obj/item/chest_key
+	name = "Mann-brand Necropolis Chest Key"
```

Comment:
its funny but it probably can't be called Mann Co

Reply:
why not? we have xenomorphs reffered to as xenomorphs